hypha/apply/funds/forms.py

```diff
@@ -408,3 +408,28 @@ def __init__(self, *args, **kwargs):
         kwargs.pop('user')
         super().__init__(*args, **kwargs)
         self.fields['meta_terms'].queryset = MetaTerm.get_root_descendants().exclude(depth=2)
+
+
+class CreateReminderForm(forms.ModelForm):
+    submission = forms.ModelChoiceField(
+        queryset=ApplicationSubmission.objects.filter(),
+        widget=forms.HiddenInput(),
+    )
+    time = forms.DateTimeField(input_formats=['%Y-%m-%dT%H:%M'])
```

Comment:
I also noticed the time input does not have a good UI. So, we tried to find a DateTime picker widget in the project. But looks like the DateTime form field on application forms also uses normal widget with modification to divide date and time into [two](https://github.com/OpenTechFund/hypha/blob/0986c35f9591347f614b21805b3306e4c11033ce/hypha/apply/stream_forms/blocks.py#L362) fields.

Looks like we can create a new issue which can be picked just after reminder feature. So, we can update all places to use a new widget like reminders, application DateTime form field etc.

Reply:
Sounds good, lets handle the widget in a separate issue and there implement it system wide.

Reply:
@frjo 

Django default format does not include `'%Y-%m-%dT%H:%M'` format.
https://docs.djangoproject.com/en/3.0/ref/settings/#datetime-input-formats
```python
[
    '%Y-%m-%d %H:%M:%S',     # '2006-10-25 14:30:59'
    '%Y-%m-%d %H:%M:%S.%f',  # '2006-10-25 14:30:59.000200'
    '%Y-%m-%d %H:%M',        # '2006-10-25 14:30'
    '%Y-%m-%d',              # '2006-10-25'
    '%m/%d/%Y %H:%M:%S',     # '10/25/2006 14:30:59'
    '%m/%d/%Y %H:%M:%S.%f',  # '10/25/2006 14:30:59.000200'
    '%m/%d/%Y %H:%M',        # '10/25/2006 14:30'
    '%m/%d/%Y',              # '10/25/2006'
    '%m/%d/%y %H:%M:%S',     # '10/25/06 14:30:59'
    '%m/%d/%y %H:%M:%S.%f',  # '10/25/06 14:30:59.000200'
    '%m/%d/%y %H:%M',        # '10/25/06 14:30'
    '%m/%d/%y',              # '10/25/06'
]
```

We need to keep `input_formats` for now because `input type=datetime-local` always has `T` separator for date and time](https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats#Local_date_and_time_strings).

Using the DateTime widget will allow us to not use `input_formats` in Django Forms.

Reply:
A space is valid instead of a "T". Users will stumble on writing dates manually as it is, adding the need for a "T" in the middle will befuddle them.

Reply:
I see, supported in Chrome and Edge but not Safari and Firefox as of now. https://caniuse.com/#search=datetime-local

In Safari users need to manually fill in the date. Filling in "2020-03-20 12:00" works no problem if "input_formats" is removed. For Chrome it seems to make no differens, I get the same UI and the same value is saved in the DB regardless if "input_formats" is there or not.

Reply:
@frjo Thanks for the `caniuse` link. In chrome, the reminder is not saved if `input_formats` is removed. Page refreshes but when we click on the reminder button, validation error displays in the modal. Looks like we should use a datepicker widget as part of this PR to fix all these browser compatibility issues.

Also, add validation for DateTime so a user can pick only future datetime.

<img width="687" alt="Screen Shot 2020-03-20 at 8 37 10 PM" src="https://user-images.githubusercontent.com/4817493/77176824-9f8e0e00-6aea-11ea-9fd7-15ae7bf35c7e.png">

Reply:
It might be that our browsers have different locales as well.

Could you please test with `input_formats=['%Y-%m-%d %H:%M']` (a space instead of T)

Reply:
Getting validation error with this input format as well `'%Y-%m-%d %H:%M'`.

Reply:
Looks like we override datetime template with input type `datetime-local` https://github.com/OpenTechFund/hypha/blob/master/hypha/templates/django/forms/widgets/datetime.html. Do you know the context of overriding it?

Reply:
>Normalized datetime strings always use the letter "T" to separate the date and the time, and the time portion of the string is as short as possible. - [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Date_and_time_formats#Local_date_and_time_strings)

Reply:
Okay, I will look out for DateTime picker and see if I can use it in our project.